movement-test/src/main.c

```diff
@@ -11,20 +14,42 @@ UINT8 PLAYER_SPRITE_ANIM[] = {
    4,    0, 0, 1, 1,     4, 6,  4,   6,   // Up
    4,    0, 0, 0, 0,     8, 10, 8, 10,   // Right
    4,    1, 1, 1, 1,     8, 10, 8, 10,   // Left
+   4,    0, 0, 0, 0,     16, 16, 16, 16,   // dash down
+   4,    0, 0, 0, 0,     12, 12, 12, 12,   // dash up
+   4,    0, 0, 0, 0,     14, 14, 14, 14,   // right
+   4,    1, 1, 1, 1,     14, 14, 14, 14,   // Left
 };
 
 // Offset of the animation of each direction in the global animation data
 #define PLAYER_DIRECTION_DOWN  0
 #define PLAYER_DIRECTION_UP    9
 #define PLAYER_DIRECTION_RIGHT 18
 #define PLAYER_DIRECTION_LEFT  27
+#define PLAYER_DASH_DIRECTION_DOWN 36
+#define PLAYER_DASH_DIRECTION_UP 45
+#define PLAYER_DASH_DIRECTION_RIGHT 54
+#define PLAYER_DASH_DIRECTION_LEFT 63
 
 // Variables containing player state
-UINT8 player_x;
-UINT8 player_y;
+RECTANGLE player;
+RECTANGLE new_player;
 UINT8 player_direction;
 UINT8 player_animation_frame;
-UINT8 is_player_walking;
+
+#define SPRITE_OFFSET_X 8
+#define SPRITE_OFFSET_Y 8  // note: set at 8 even though sprites are 8px tall because of the perspective
+
+#define TILE_SIZE_PX 8
+#define MAX_MOVE_X (TILE_SIZE_PX - 1)
+#define MAX_MOVE_Y (TILE_SIZE_PX - 1)
+
+#define COLDOWN_DASH 200            //cooldown of the dash (in number of frame)  (must be < 255)
```

Comment:
typo: CO**O**LDOWN

Reply:
Also, should probably be a global variable, in case we want to change that in-game